File: api/python/Grpcs/TextFilterGrpcProvider.py
#!/usr/bin/env python
# -*- coding:utf-8 -*-
# ImageFilterAsyncProvider.py
# 2021, Lin Zhijun, https://github.com/toolgood/ToolGood.TextFilter.Api
# MIT Licensed 

import grpc
from Grpcs.GrpcBase import textFilter_pb2
from Grpcs.GrpcBase import textFilter_pb2_grpc


class TextFilterGrpcProvider():
    def __init__(self,textFilterConfig):
        self.__textFilterConfig=textFilterConfig

    def TextFilter(self,txt:str,skipBidi:bool,onlyPosition:bool):
        """
        @txt 需要检测的文本

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字
        """
        channel = grpc.insecure_channel(self.__textFilterConfig.GetGrpcHost())
        stub = textFilter_pb2_grpc.TextFilterGrpcStub(channel)
        response = stub.TextFilter(textFilter_pb2.TextFindAllGrpcRequest(txt=txt, skipBidi=skipBidi,onlyPosition=onlyPosition))
        return response


    def TextReplace(self,txt:str,replaceChar:str,reviewReplace:bool,contactReplace:bool,skipBidi:bool,onlyPosition:bool):
        """
        @txt 需要检测的文本

        @replaceChar 替换符号, 默认 星号

        @reviewReplace 人工审核替换，默认true

        @contactReplace 联系字符串替换，默认true

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字

        """
        channel = grpc.insecure_channel(self.__textFilterConfig.GetGrpcHost())
        stub = textFilter_pb2_grpc.TextFilterGrpcStub(channel)
        
        replaceCh=ord('*')
        if len(replaceChar)>0:
            replaceCh=ord(replaceChar[0])
        response = stub.TextReplace(textFilter_pb2.TextReplaceGrpcRequest(txt=txt,replaceChar=replaceCh,reviewReplace=reviewReplace,contactReplace=contactReplace, skipBidi=skipBidi,onlyPosition=onlyPosition))
        return response

    def HtmlFilter(self,txt:str,skipBidi:bool,onlyPosition:bool):
        """
        @txt 需要检测的文本
        @skipBidi 是否跳过 Bidi 字符，默认 false
        @onlyPosition 只显示位置，不显示匹配文字
        """
        channel = grpc.insecure_channel(self.__textFilterConfig.GetGrpcHost())
        stub = textFilter_pb2_grpc.TextFilterGrpcStub(channel)
        response = stub.HtmlFilter(textFilter_pb2.TextFindAllGrpcRequest(txt=txt, skipBidi=skipBidi,onlyPosition=onlyPosition))
        return response
 

    def HtmlReplace(self,txt:str,replaceChar:str,reviewReplace:bool,contactReplace:bool,skipBidi:bool,onlyPosition:bool):
        """
        @txt 需要检测的文本

        @replaceChar 替换符号, 默认 星号

        @reviewReplace 人工审核替换，默认true

        @contactReplace 联系字符串替换，默认true

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字

        """
        channel = grpc.insecure_channel(self.__textFilterConfig.GetGrpcHost())
        stub = textFilter_pb2_grpc.TextFilterGrpcStub(channel)
        replaceCh=ord('*')
        if len(replaceChar)>0:
            replaceCh=ord(replaceChar[0])
        response = stub.HtmlReplace(textFilter_pb2.TextReplaceGrpcRequest(txt=txt,replaceChar=replaceCh,reviewReplace=reviewReplace,contactReplace=contactReplace, skipBidi=skipBidi,onlyPosition=onlyPosition))
        return response

    def JsonFilter(self,txt:str,skipBidi:bool,onlyPosition:bool):
        """
        @txt 需要检测的文本

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字
        """
        channel = grpc.insecure_channel(self.__textFilterConfig.GetGrpcHost())
        stub = textFilter_pb2_grpc.TextFilterGrpcStub(channel)
        response = stub.JsonFilter(textFilter_pb2.TextFindAllGrpcRequest(txt=txt, skipBidi=skipBidi,onlyPosition=onlyPosition))
        return response

    def JsonReplace(self,txt:str,replaceChar:str,reviewReplace:bool,contactReplace:bool,skipBidi:bool,onlyPosition:bool):
        """
        @txt 需要检测的文本

        @replaceChar 替换符号, 默认 星号

        @reviewReplace 人工审核替换，默认true

        @contactReplace 联系字符串替换，默认true

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字

        """
        channel = grpc.insecure_channel(self.__textFilterConfig.GetGrpcHost())
        stub = textFilter_pb2_grpc.TextFilterGrpcStub(channel)
        replaceCh=ord('*')
        if len(replaceChar)>0:
            replaceCh=ord(replaceChar[0])
        response = stub.JsonReplace(textFilter_pb2.TextReplaceGrpcRequest(txt=txt,replaceChar=replaceCh,reviewReplace=reviewReplace,contactReplace=contactReplace, skipBidi=skipBidi,onlyPosition=onlyPosition))
        return response
 
    def MarkdownFilter(self,txt:str,skipBidi:bool,onlyPosition:bool):
        """
        @txt 需要检测的文本

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字
        """
        channel = grpc.insecure_channel(self.__textFilterConfig.GetGrpcHost())
        stub = textFilter_pb2_grpc.TextFilterGrpcStub(channel)
        response = stub.MarkdownFilter(textFilter_pb2.TextFindAllGrpcRequest(txt=txt, skipBidi=skipBidi,onlyPosition=onlyPosition))
        return response

    def MarkdownReplace(self,txt:str,replaceChar:str,reviewReplace:bool,contactReplace:bool,skipBidi:bool,onlyPosition:bool):
        """
        @txt 需要检测的文本

        @replaceChar 替换符号, 默认 星号

        @reviewReplace 人工审核替换，默认true

        @contactReplace 联系字符串替换，默认true

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字

        """
        channel = grpc.insecure_channel(self.__textFilterConfig.GetGrpcHost())
        stub = textFilter_pb2_grpc.TextFilterGrpcStub(channel)
        replaceCh=ord('*')
        if len(replaceChar)>0:
            replaceCh=ord(replaceChar[0])
        response = stub.MarkdownReplace(textFilter_pb2.TextReplaceGrpcRequest(txt=txt,replaceChar=replaceCh,reviewReplace=reviewReplace,contactReplace=contactReplace, skipBidi=skipBidi,onlyPosition=onlyPosition))
        return response

File: api/golang/ToolGood/TextFilter/Api/Grpcs/TextFilterGrpcProvider.go
package Grpcs

import (
	"log"

	pb "./GrpcBase"
	"golang.org/x/net/context"
	"google.golang.org/grpc"

	. "../Datas/Requests"
	. "../Datas/Texts"
)

type TextFilterGrpcProvider struct {
	grpcHost string
}

func NewTextFilterGrpcProvider(grpcHost string) *TextFilterGrpcProvider {
	result := &TextFilterGrpcProvider{}
	result.grpcHost = grpcHost
	return result
}

func (this *TextFilterGrpcProvider) TextFilter(request *TextFilterRequest) *TextFilterResult {
	conn, err := grpc.Dial(this.grpcHost, grpc.WithInsecure())
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	t := pb.NewTextFilterGrpcClient(conn)

	reqBody := new(pb.TextFindAllGrpcRequest)
	reqBody.Txt = request.Txt
	reqBody.SkipBidi = request.SkipBidi
	reqBody.OnlyPosition = request.OnlyPosition

	tr, err := t.TextFilter(context.Background(), reqBody)
	if err != nil {
		log.Fatalf("did not connect: %v", err)
		return nil
	}
	result := this.createTextFilterResult(tr)
	defer conn.Close()
	return result
}
func (this *TextFilterGrpcProvider) HtmlFilter(request *TextFilterRequest) *TextFilterResult {
	conn, err := grpc.Dial(this.grpcHost, grpc.WithInsecure())
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	t := pb.NewTextFilterGrpcClient(conn)

	reqBody := new(pb.TextFindAllGrpcRequest)
	reqBody.Txt = request.Txt
	reqBody.SkipBidi = request.SkipBidi
	reqBody.OnlyPosition = request.OnlyPosition

	tr, err := t.HtmlFilter(context.Background(), reqBody)
	if err != nil {
		log.Fatalf("did not connect: %v", err)
		return nil
	}
	result := this.createTextFilterResult(tr)
	defer conn.Close()
	return result
}
func (this *TextFilterGrpcProvider) JsonFilter(request *TextFilterRequest) *TextFilterResult {
	conn, err := grpc.Dial(this.grpcHost, grpc.WithInsecure())
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	t := pb.NewTextFilterGrpcClient(conn)

	reqBody := new(pb.TextFindAllGrpcRequest)
	reqBody.Txt = request.Txt
	reqBody.SkipBidi = request.SkipBidi
	reqBody.OnlyPosition = request.OnlyPosition

	tr, err := t.JsonFilter(context.Background(), reqBody)
	if err != nil {
		log.Fatalf("did not connect: %v", err)
		return nil
	}
	result := this.createTextFilterResult(tr)
	defer conn.Close()
	return result
}
func (this *TextFilterGrpcProvider) MarkdownFilter(request *TextFilterRequest) *TextFilterResult {
	conn, err := grpc.Dial(this.grpcHost, grpc.WithInsecure())
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	t := pb.NewTextFilterGrpcClient(conn)

	reqBody := new(pb.TextFindAllGrpcRequest)
	reqBody.Txt = request.Txt
	reqBody.SkipBidi = request.SkipBidi
	reqBody.OnlyPosition = request.OnlyPosition

	tr, err := t.MarkdownFilter(context.Background(), reqBody)
	if err != nil {
		log.Fatalf("did not connect: %v", err)
		return nil
	}
	result := this.createTextFilterResult(tr)
	defer conn.Close()
	return result
}

func (this *TextFilterGrpcProvider) TextReplace(request *TextReplaceRequest) *TextReplaceResult {
	conn, err := grpc.Dial(this.grpcHost, grpc.WithInsecure())
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	t := pb.NewTextFilterGrpcClient(conn)

	reqBody := new(pb.TextReplaceGrpcRequest)
	reqBody.Txt = request.Txt
	reqBody.ReplaceChar = 42 //*
	if len(request.ReplaceChar) > 0 {
		byteArray := []rune(request.ReplaceChar)
		reqBody.ReplaceChar = uint32(byteArray[0])
	}
	reqBody.ReviewReplace = request.ReviewReplace
	reqBody.ContactReplace = request.ContactReplace
	reqBody.SkipBidi = request.SkipBidi
	reqBody.OnlyPosition = request.OnlyPosition

	tr, err := t.TextReplace(context.Background(), reqBody)
	if err != nil {
		log.Fatalf("did not connect: %v", err)
		return nil
	}
	result := this.createTextReplaceResult(tr)
	defer conn.Close()
	return result
}

func (this *TextFilterGrpcProvider) HtmlReplace(request *TextReplaceRequest) *TextReplaceResult {
	conn, err := grpc.Dial(this.grpcHost, grpc.WithInsecure())
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	t := pb.NewTextFilterGrpcClient(conn)

	reqBody := new(pb.TextReplaceGrpcRequest)
	reqBody.Txt = request.Txt
	reqBody.ReplaceChar = 42 //*
	if len(request.ReplaceChar) > 0 {
		byteArray := []rune(request.ReplaceChar)
		reqBody.ReplaceChar = uint32(byteArray[0])
	}
	reqBody.ReviewReplace = request.ReviewReplace
	reqBody.ContactReplace = request.ContactReplace
	reqBody.SkipBidi = request.SkipBidi
	reqBody.OnlyPosition = request.OnlyPosition

	tr, err := t.HtmlReplace(context.Background(), reqBody)
	if err != nil {
		log.Fatalf("did not connect: %v", err)
		return nil
	}
	result := this.createTextReplaceResult(tr)
	defer conn.Close()
	return result
}

func (this *TextFilterGrpcProvider) JsonReplace(request *TextReplaceRequest) *TextReplaceResult {
	conn, err := grpc.Dial(this.grpcHost, grpc.WithInsecure())
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	t := pb.NewTextFilterGrpcClient(conn)

	reqBody := new(pb.TextReplaceGrpcRequest)
	reqBody.Txt = request.Txt
	reqBody.ReplaceChar = 42 //*
	if len(request.ReplaceChar) > 0 {
		byteArray := []rune(request.ReplaceChar)
		reqBody.ReplaceChar = uint32(byteArray[0])
	}
	reqBody.ReviewReplace = request.ReviewReplace
	reqBody.ContactReplace = request.ContactReplace
	reqBody.SkipBidi = request.SkipBidi
	reqBody.OnlyPosition = request.OnlyPosition

	tr, err := t.JsonReplace(context.Background(), reqBody)
	if err != nil {
		log.Fatalf("did not connect: %v", err)
		return nil
	}
	result := this.createTextReplaceResult(tr)
	defer conn.Close()
	return result
}

func (this *TextFilterGrpcProvider) MarkdownReplace(request *TextReplaceRequest) *TextReplaceResult {
	conn, err := grpc.Dial(this.grpcHost, grpc.WithInsecure())
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	t := pb.NewTextFilterGrpcClient(conn)

	reqBody := new(pb.TextReplaceGrpcRequest)
	reqBody.Txt = request.Txt
	reqBody.ReplaceChar = 42 //*
	if len(request.ReplaceChar) > 0 {
		byteArray := []rune(request.ReplaceChar)
		reqBody.ReplaceChar = uint32(byteArray[0])
	}
	reqBody.ReviewReplace = request.ReviewReplace
	reqBody.ContactReplace = request.ContactReplace
	reqBody.SkipBidi = request.SkipBidi
	reqBody.OnlyPosition = request.OnlyPosition

	tr, err := t.MarkdownReplace(context.Background(), reqBody)
	if err != nil {
		log.Fatalf("did not connect: %v", err)
		return nil
	}
	result := this.createTextReplaceResult(tr)
	defer conn.Close()
	return result
}

func (this *TextFilterGrpcProvider) createTextFilterResult(response *pb.TextFindAllGrpcReply) *TextFilterResult {
	result := &TextFilterResult{}
	result.Code = response.GetCode()
	result.RiskCode = response.GetRiskCode()
	result.RequestId = response.GetRequestId()
	result.Message = response.GetMessage()
	result.RiskLevel = response.GetRiskLevel()
	result.SentimentScore = response.GetSentimentScore()
	details := make([]TextFilterDetailItem, len(response.Details))
	for index, r := range response.Details {
		item := &TextFilterDetailItem{}
		item.Position = r.GetPosition()
		item.RiskCode = r.GetRiskCode()
		item.RiskLevel = r.GetRiskLevel()
		item.Text = r.GetText()
		details[index] = *item
	}
	result.Details = details

	contactItems := make([]TextFilterContactItem, len(response.Contacts))
	for index, r := range response.Contacts {
		item := &TextFilterContactItem{}
		item.ContactType = r.GetContactType()
		item.ContactString = r.GetContactString()
		item.Position = r.GetPosition()
		contactItems[index] = *item
	}
	result.Contacts = contactItems
	return result
}

func (this *TextFilterGrpcProvider) createTextReplaceResult(response *pb.TextReplaceGrpcReply) *TextReplaceResult {
	result := &TextReplaceResult{}
	result.Code = response.GetCode()
	result.RequestId = response.GetRequestId()
	result.Message = response.GetMessage()
	result.RiskLevel = response.GetRiskLevel()
	details := make([]TextFilterDetailItem, len(response.Details))
	for index, r := range response.Details {
		item := &TextFilterDetailItem{}
		item.Position = r.GetPosition()
		item.RiskCode = r.GetRiskCode()
		item.RiskLevel = r.GetRiskLevel()
		item.Text = r.GetText()
		details[index] = *item
	}
	result.Details = details

	return result
}
